implementation.md

```diff
@@ -0,0 +1,326 @@
+1. Create a new interface, _R_, extending `InputRepository<`_V_`>`.
+1. Create a new class that extends `BaseInputFormatDefinition<`_V_`,`_R_`> and
+provides those types to the constructor as well as a name that will be used in
+the `Input` instruction. This class must be annotated with
+`@MetaInfServices(InputFormatDefinition)`.
+1. Create new classes that provide data which implement _R_ and are annotated
+with `@MetaInfServices(`_R_`)`.
+
+### Constants
+A constant is a value that can be generate with no input. It need not actually
+be constant (_e.g._, `now` is a constant). The Shesmu compiler will arbitrarily
+copy the value of a constant, so a constant should be side-effect free.
+
+1. Create a class that implements `ConstantSource`.
+1. Annotate this class with `@MetaInfServices`.
+1. Include a no-arguments constructor for the `ConstantSource` class.
+1. Return a stream of `Constant` objects for the constants to be created.
+
+There are a number of convience methods to create constants for values. If
+something more complicated is desired, extend the `Constant` class and write
+arbitrary bytecode in the `load` method. The code must not change any value on
+the stack and must leave exactly one value of the correct type on the stack.
+
+### Functions
+A function is a transformation of input data. It matches with a call to a
+method. Since there is no user-defined error-handling, these functions should
+not throw. Also, since Shesmu has no null values, they should not return null
+on an error.
+
+To provide functions:
+
+1. Create a class taht implements `FunctionRepository`.
+1. Annotate this class with `@MetaInfServices`.
+1. Include a no-arguments constructor for the `FunctionRepository` class.
+1. Return a stream of `FunctionDefintion` objects for the functions to be created.
+
+There are a few methods to generate a `FunctionDefinition`:
+
+1. Create a class that implements `FunctionDefinition` and write arbitrary
+bytecode.
+1. Use `FunctionDefinition.staticMethod` to create a binding for a public
+static method in a class.
+1. Use `FunctionForInstance.bind` to create a binding for a method in an
+instance of a class (a closure).
+
+In all cases, the Java types for the parameters must match the Shesmu types
+provided. The order of the arguments must match. If they do not match, errors
+will occur during compilation of an olive.
+
+When writing byte code, the arguments will be on the stack in order.
+
+When binding an instance, a `MethodHandles.Lookup` object must be provided.
+There are two ways to obtain one
+[`publicLookup()`](https://docs.oracle.com/javase/8/docs/api/java/lang/invoke/MethodHandles.html#publicLookup--),
+which can only bind to public methods, and
+[`lookup()`](https://docs.oracle.com/javase/8/docs/api/java/lang/invoke/MethodHandles.html#lookup--),
+which can bind to any method that is visible from the caller.
+
+### Actions
+Actions are the most likely thing that must be added to Shesmu, but they are
```

Comment:
This isn't true anymore.

Reply:
No? I think of someone were to pick this up off the shelf, it probably isn't going to work with their workflow engine, even if we collect a bunch of those.

Reply:
In the README you say that all kinds of plugins are likely going to need to be implemented for someone to use it locally, so it's not the most likely thing. Maybe the most common thing after deployment?

Reply:
So I did! :dizzy_face: